soc/riscv/riscv-ite/it8xxx2/__arithmetic.S

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2022 ITE Corporation.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * When the 'M' extension is disabled, compiler can not recognize div/mul
+ * instructions. So mul/div instructions in the below integer arithmetic
+ * routines are hard coded by opcodes.
+ */
+
+/* signed 32 bit multiplication */
+	.text
```

Comment:
Would it make sense to put these functions in `__ram_code`? They're very small and reducing I-cache pressure seems like a significant win for these functions.

I think code size could be reduced a little too, if the `nop`s were replaced with jumps to a shared return:

```
__mulsi3:
    .word 0x02b50533
    j arithmetic_ret

__divsi3:
    .word 0x02b54533
    j arithmetic_ret

arithmetic_ret:
    ret
```

This would save one instruction per function, but might not be worthwhile if the jump requires a pipeline stall.

Reply:
> Would it make sense to put these functions in `__ram_code`? They're very small and reducing I-cache pressure seems like a significant win for these functions.
> 
Done. Let's put these functions in the __ram_code section.

> I think code size could be reduced a little too, if the `nop`s were replaced with jumps to a shared return:
> 
> ```
> __mulsi3:
>     .word 0x02b50533
>     j arithmetic_ret
> 
> __divsi3:
>     .word 0x02b54533
>     j arithmetic_ret
> 
> arithmetic_ret:
>     ret
> ```
> 
> This would save one instruction per function, but might not be worthwhile if the jump requires a pipeline stall.

Yep, your change will reduce code size.
Since SA recommends the workaround sequence, I would prefer to keep the code as it was.

Reply:
What's "SA"?

Reply:
System Analysis.